src/migrations/20240702000000-remove_national_center_ars.js

```diff
@@ -0,0 +1,553 @@
+          USING nextsteps_to_delete
+          WHERE id = nsid
+          RETURNING
+          id
+        )
+        SELECT id FROM deletes
+        ;
+
+
+        -- Create the ARO deletion list
+        DROP TABLE IF EXISTS aros_to_delete;
+        CREATE TEMP TABLE aros_to_delete
+        AS
+        SELECT
+          id aroid,
+          "objectiveId" oid
+        FROM "ActivityReportObjectives"
+        JOIN ars_to_delete
+          ON "activityReportId" = arid
+        ;
+        -- Remove ARO link records: -------------
+        DROP TABLE IF EXISTS deleted_activityreportobjectivefiles;
+        CREATE TEMP TABLE deleted_activityreportobjectivefiles AS
+        WITH deletes AS (
+          DELETE FROM "ActivityReportObjectiveFiles"
+          USING aros_to_delete
+          WHERE "activityReportObjectiveId" = aroid
+          RETURNING
+          id,
+          "fileId" fid
+        )
+        SELECT id, fid FROM deletes
+        ;
+        DROP TABLE IF EXISTS deleted_activityreportobjectiveresources;
+        CREATE TEMP TABLE deleted_activityreportobjectiveresources AS
+        WITH deletes AS (
+          DELETE FROM "ActivityReportObjectiveResources"
+          USING aros_to_delete
+          WHERE "activityReportObjectiveId" = aroid
+          RETURNING
+          id,
+          "resourceId" resourceid
+        )
+        SELECT id, resourceid FROM deletes
+        ;
+        DROP TABLE IF EXISTS deleted_activityreportobjectivetopics;
+        CREATE TEMP TABLE deleted_activityreportobjectivetopics AS
+        WITH deletes AS (
+          DELETE FROM "ActivityReportObjectiveTopics"
+          USING aros_to_delete
+          WHERE "activityReportObjectiveId" = aroid
+          RETURNING
+          id
+        )
+        SELECT id FROM deletes
+        ;
+        -- remove AROs -------------------
+        DROP TABLE IF EXISTS deleted_aros;
+        CREATE TEMP TABLE deleted_aros AS
+        WITH deletes AS (
+          DELETE FROM "ActivityReportObjectives"
+          USING aros_to_delete
+          WHERE id = aroid
+          RETURNING
+          id,
+          "objectiveId" oid
+        )
+        SELECT id, oid FROM deletes
+        ;
+
+        -- Create the orphaned Objective deletion list
+        DROP TABLE IF EXISTS objectives_to_delete;
+        CREATE TEMP TABLE objectives_to_delete
+        AS
+        SELECT DISTINCT oid
+        FROM deleted_aros
+        EXCEPT
+        SELECT DISTINCT "objectiveId"
+        FROM "ActivityReportObjectives"
+        ;
+        
+        -- remove Objectives -------------------
+        DROP TABLE IF EXISTS deleted_objectives;
+        CREATE TEMP TABLE deleted_objectives AS
+        WITH deletes AS (
+          DELETE FROM "Objectives"
+          USING objectives_to_delete
+          WHERE id = oid
+          RETURNING
+          id,
+          "goalId" gid,
+          "objectiveTemplateId" otid
+        )
+        SELECT id, gid, otid FROM deletes
+        ;
```

Comment:
ObjectiveColaborators must be deleted first

Reply:
OnApprovedAR and OnAR need to be updated for any objectives not deleted

Reply:
Added ObjectiveCollaborators. The `onApprovedAR` and `onAR` are at the end